optimum_benchmark/benchmarks/training.py

```diff
@@ -181,12 +207,21 @@ def run(self, backend: Backend) -> None:
                 f"Training benchmark not implemented for task {backend.task}."
                 "Please submit a PR to add support for this task."
             )
+        return training_dataset
+
+    def run(self, backend: "Backend") -> None:
+        LOGGER.info("Running training benchmark")
+
+        self.training_dataset = self.generate_dataset(backend)
 
-        backend.prepare_for_training(
-            training_dataset=self.training_dataset,
-            training_arguments=self.training_arguments,
-        )
-        results = backend.train().metrics
+        if backend.config.name == "pytorch":
+            results = backend.run_pytorch_training(training_config=self.config, training_arguments=self.training_arguments, training_dataset=self.training_dataset)
```

Comment:
Passing `self` arguments to an external class method is slightly inelegant.

Reply:
yes that's why I would prefer dispatching on `use_ddp` here.
I also think it'd be better to decouple vanilla training (through backend.train) which is a homogenous feature across backends, from ddp training.

Reply:
I don't see how dispatching on `use_ddp` would change the way arguments are passed, could you explain?

I don't understand why we should inject DDP logic in TrainingBenchmark given that it is specific to PyTorch (or at least, currently untested with any other backend).